### ISSUE TITLE
Replace file() with open() in twisted/web

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8519]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8519|
|Type|enhancement|
|Created|2016-06-28 06:17:16Z|
|Branch|https://github.com/twisted/twisted/tree/8519-rodrigc-file|

In Python 3, file() is gone (see https://docs.python.org/3/whatsnew/3.0.html#builtins ), and needs to be replaced with open()
which works on Python 2 and 3

<details><summary>Searchable metadata</summary>

```
trac-id__8519 8519
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8519_rodrigc_file 8519-rodrigc-file
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1467094636229546 1467094636229546
changetime__1468800601420573 1468800601420573
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/243

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_2"></a>|@Lukasa set owner to @rodrigc|
|-|-|

Cool, twistedchecker is still being dumb but this is otherwise good. Merge when ready.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc commented|
|-|-|

Due to coverage issues, I had to remove the part of my patch which changed the _cbPartialTest() function.  This is because the testPartial() test is always being skipped, and receiving no coverage.  Changing something which is receiving no coverage
is causing codecov to complain.  Please review again.

Also, please look at [ticket:8530] where I propose to delete the testPartial() tests, which is essentially dead code.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)|@rodrigc removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban set owner to @rodrigc|
|-|-|

Look good. Thanks! Please merge.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|GitHub set status to `closed`|
|-|-|

In changeset a14adae02bd763ce9c2613ca954ceecf6e745fb4
```
#!CommitTicketReference repository="" revision="a14adae02bd763ce9c2613ca954ceecf6e745fb4"
Merge pull request #243 from twisted/8519-rodrigc-file

Author: rodrigc
Reviewer: alex, Lukasa, adiroiban
Fixes: #8519

Replace file() with open() in twisted/web
```
